# **📊 1: How many locations are in the game?**

In [0]:
%sql
SELECT COUNT (DISTINCT *) AS total_locations 
FROM eldenringcatalog.gold.dim_locations;

total_locations 281

In [0]:
%sql
SELECT location_id, location_name 
FROM eldenringcatalog.gold.dim_locations
LIMIT 10 ;

location_id location_name 0 Abandoned Ailing Village 1 Artist's Shack (Gravesite Plain) 2 Belurat Gaol 3 Belurat Tower Settlement 4 Castle Ensis 5 Church of Benediction 6 Church of Consolation 7 Dragon's Pit 8 Elder's Hovel 9 Ellac Greatbridge

**Based on the data we currently have, the eldenringcatalog.gold.dim_locations table contains 281 distinct locations from Elden Ring. These include well-known areas such as Abandoned Ailing Village, Artist’s Shack (Gravesite Plain), Belurat Gaol, Castle Ensis, and Ellac Greatbridge. This dataset provides broad coverage of the world, but future updates may refine or expand the list as more official information becomes available.**

# **📊 2: Load NPCs and count total**

In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_npcs 
FROM eldenringcatalog.gold.dim_npcs;

total_npcs 62

In [0]:
%sql
SELECT npc_name, role, is_dlc 
FROM eldenringcatalog.gold.dim_npcs 
LIMIT 30;

npc_name role is_dlc Abandoned Merchant Siofra River General Goods Shop 0 Aeonian Spirit Benevolent 0 Aeonian Swamp Spirit Benevolent 0 Albinauric Village Spirit Benevolent 0 Albus Gives Key Item 0 Sir Ansbach Pureblood Knight Captain (former), NPC Quest Related 1 Ashen Spirit Benevolent 0 Blackguard Big Boggart Blackguard 0 Blaidd the Half-Wolf Quest NPC and summon 0 Bloody Finger Hunter Yura Quest NPC and Summonable Cooperator 0 Boc the Seamster Garments Adjuster andQuest NPC 0 Brother Corhyn Merchant and Quest NPC 0 Carian Manor Spirit Benevolent 0 Church of Pilgrimage Spirit Spirit NPC 0 Count Ymir, High Priest Merchant 1 D, Beholder of Death Quest NPC and Summon 0 D, Hunter of the Dead Quest NPC and Summon 0 Deathtouched Spirit Spirit NPC 0 Knight Diallos Quest NPC 0 Dragon Communion Priestess Questline Related 1 Dryleaf Dane Mini-Boss 1 Dung Eater Quest NPC, Invader, and Summon 0 Eclipse Spirit Spirit NPC 0 Edgar Morne Castellan, Quest NPC 0 Enia Finger Reader 0 Ensha of the Royal Remains NPC Invader 0 Erdtree Sanctuary Spirit Benevolent 0 Fia, Deathbed Companion Quest NPC 0 Finger Maiden Therolina NPC Summon 0 Fire Knight Queelign NPC Invader, NPC 1

**Based on the data available in our dataset, Elden Ring contains 62 unique NPCs, covering a wide range of roles such as quest-related characters, merchants, invaders, benevolent spirits, and summonable allies. The preview shows a diverse mix, including key figures like Blaidd, Dung Eater, Fia, Enia, and various spirits and merchants. The dataset also indicates which NPCs belong to the DLC, such as Sir Ansbach, Count Ymir, and Dryleaf Dane. These insights reflect the current version of the data, and any missing or newly added NPCs may appear in future updates.**

# **📊 3: NPC roles distribution**

In [0]:
%sql
SELECT 
    role,
    COUNT( DISTINCT *) AS npc_count
FROM eldenringcatalog.gold.dim_npcs
GROUP BY role
ORDER BY npc_count DESC;

role npc_count Quest NPC 8 Merchant 5 Quest NPC and Summon 4 General Goods Shop 3 Quest NPC and Summonable Cooperator 2 Guide and Quest NPC 1 Mini-Boss 1 Spirit-Tuner 1 NPC and Summonable Cooperator 1 Guide and Lore NPC 1 NPC Quest Related 1 Gives Key Item 1 Blacksmith & Quest NPC 1 Drake Warrior, NPC Quest Related 1 Elder Sage 1 Boss, Merchant, Quest NPC, Summon 1 Blackguard 1 Maiden 1 Quest NPC and Merchant 1 The Storyteller 1 Last Carian Queen 1 Scout 1 Special Trader 1 Merchant, Quest NPC, and Summon 1 Merchant, NPC Quest Related 1 Blacksmith 1 Garments Adjuster andQuest NPC 1 Questline Related 1 Quest NPC and summon 1 Hornsent Survivor 1 Ash of War Merchant 1 Quest NPC, Invader, and Summon 1 Servant of Count Ymir 1 Next Rightful Ruler of Limgrave 1 Amicable NPC, Invader NPC 1 Finger Reader 1 Morne Castellan, Quest NPC 1 Miquella's Followers Leader 1 Pureblood Knight Captain (former), NPC Quest Related 1 NPC Invader, NPC 1 Merchant and Quest NPC 1 Main NPC - Affects Endings 1 Sorcery Student 1 NPC and Merchant 1 Cooperator, Questline Related 1 NPC Summon 0 First Demigod to die 0 Ruler of The Lands Between 0 NPC 0 Messenger of The Greater Will 0 Spirit NPC 0 Summon and Boss 0 NPC Invader 0 Benevolent 0 null 0 Ending of Millicent questline 0 God Greater Will Manifestation 0

**The NPC roles in Elden Ring are highly varied, with Quest NPCs being the most common (8), followed by Merchants (5) and hybrid roles like Quest NPC and Summon (4). Many roles appear only once, reflecting unique or story-specific characters, such as Blacksmith & Quest NPC, The Storyteller, or Miquella's Followers Leader. Several entries have zero counts or inconsistent labeling, highlighting the need for standardization of role categories to improve analysis and avoid duplicates or null/ambiguous entries. This would make role-based insights clearer and more actionable.**

# **📊 4: DLC NPCs vs base game**

In [0]:
%sql
SELECT 
    CASE WHEN is_dlc = 1 THEN 'DLC' ELSE 'Base Game' END AS content_type,
    COUNT(DISTINCT *) AS npc_count
FROM eldenringcatalog.gold.dim_npcs
GROUP BY is_dlc;

content_type npc_count Base Game 48 DLC 14

**In Elden Ring, most NPCs are from the Base Game (48), while a smaller portion belongs to DLC content (14). This shows that the majority of characters players encounter are part of the core game, with DLC adding a more limited but potentially unique set of NPCs.**

# **📊 5: Load items and analyze item types**

In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_items 
FROM eldenringcatalog.gold.dim_items;

total_items 584

In [0]:
%sql
SELECT 
    item_type,
    COUNT(DISTINCT *) AS item_count
FROM eldenringcatalog.gold.dim_items
GROUP BY item_type
ORDER BY item_count DESC
limit 5;

item_type item_count Consumable 209 Cookbook 106 Material 105 Bell 60 Upgrade Material 42

**In Elden Ring, the game features 584 distinct items. The most common item types are Consumables (209), Cookbooks (106), and Materials (105), followed by Bells (60) and Upgrade Materials (42). This indicates that healing, crafting, and upgrade mechanics play a major role in gameplay.**

# **📊 6: Load location-items bridge table**

In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_relationships 
FROM eldenringcatalog.gold.bridge_location_items;

total_relationships 2937

In [0]:
%sql
SELECT * 
FROM eldenringcatalog.gold.bridge_location_items 
LIMIT 20;

location_item_key location_id item_name created_at 0 0 Broken Rune 2025-11-29T17:01:17.003Z 1 0 Golden Rune (1) 2025-11-29T17:01:17.003Z 2 0 Black Pyrefly 2025-11-29T17:01:17.003Z 3 0 Human Bone Shard 2025-11-29T17:01:17.003Z 4 0 Fly Mold 2025-11-29T17:01:17.003Z 5 0 Ailment Talisman 2025-11-29T17:01:17.003Z 6 0 Revered Spirit Ash 2025-11-29T17:01:17.003Z 7 0 Greater Potentates Cookbook (10) 2025-11-29T17:01:17.003Z 8 0 Larval Tear 2025-11-29T17:01:17.003Z 9 1 Incursion Painting 2025-11-29T17:01:17.003Z 10 2 Frozen Maggot 2025-11-29T17:01:17.003Z 11 2 Broken Rune 2025-11-29T17:01:17.003Z 12 2 Shadow Realm Rune 3 2025-11-29T17:01:17.003Z 13 2 Smithing Stone 3 2025-11-29T17:01:17.003Z 14 2 Smithing Stone 6 2025-11-29T17:01:17.003Z 15 2 Greatjar 2025-11-29T17:01:17.003Z 16 2 Glass Shard 2025-11-29T17:01:17.003Z 17 2 Innard Meat 2025-11-29T17:01:17.003Z 18 2 Thawfrost Boluses 2025-11-29T17:01:17.003Z 19 2 Silver Horn Tender 2025-11-29T17:01:17.003Z

**The bridge_location_items table in Elden Ring contains 2,937 distinct relationships linking locations to items. Each record represents a specific item found at a particular location, including its creation timestamp. For example, location 0 has items like Broken Rune, Golden Rune (1), and Black Pyrefly, while location 2 includes items such as Frozen Maggot, Smithing Stone 3, and Innard Meat. This table is essential for tracking item distribution and accessibility across the game world.**

# **📊 7: Which location has the most items?**

In [0]:
%sql
SELECT 
    l.location_name,
    l.location_id,
    COUNT(DISTINCT *) AS item_count
FROM eldenringcatalog.gold.bridge_location_items b
JOIN eldenringcatalog.gold.dim_locations l ON b.location_id = l.location_id
GROUP BY l.location_name, l.location_id
ORDER BY item_count DESC
LIMIT 20

location_name location_id item_count Leyndell, Royal Capital 230 84 Stormveil Castle 81 82 Raya Lucaria Academy 144 64 Volcano Manor 217 53 Elphael Brace of the Haligtree 263 50 Subterranean Shunning-Grounds 284 41 Siofra River 283 40 Nokron, Eternal City 280 39 Deeproot Depths 272 38 Castle Morne 90 34 Miquella's Haligtree 266 32 Ainsel River 270 29 Mohgwyn Palace 277 28 Mistwood 70 27 The Shaded Castle 200 27 Lake of Rot 276 26 Caria Manor 115 24 Jarburg 132 24 Nokstella, Eternal City 281 21 Fort of Reprimand 28 20

**The top 20 locations in Elden Ring with the most unique items show that Leyndell, Royal Capital leads with 84 unique items, followed closely by Stormveil Castle with 82 unique items and Raya Lucaria Academy with 64 unique items. Other notable locations include Volcano Manor (53 unique items), Elphael Brace of the Haligtree (50 unique items), and Subterranean Shunning-Grounds (41 unique items). This highlights that major castles, academies, and late-game areas tend to contain the densest concentration of unique collectible items.**

# **📊 8: Locations with fewest items**

In [0]:
%sql
SELECT 
    l.location_name,
    l.location_id,
    COUNT(DISTINCT *) AS item_count
FROM eldenringcatalog.gold.bridge_location_items b
JOIN eldenringcatalog.gold.dim_locations l ON b.location_id = l.location_id
GROUP BY l.location_name, l.location_id
ORDER BY item_count
LIMIT 20;

location_name location_id item_count Castle Ensis 4 0 Belurat Tower Settlement 3 0 Midra's Manse 46 0 Enir-Ilim 25 0 Shadow Keep 33 0 Sellia Evergaol 240 1 Malefactor's Evergaol 137 1 Golden Lineage Evergaol 187 1 Cuckoo's Evergaol 125 1 Forlorn Hound Evergaol 62 1 Ranni's Rise 143 1 Whipping Hut 35 1 Albinauric Rise 259 1 Oridys's Rise 100 1 Frenzy-Flaming Tower 129 1 The Four Belfries 158 1 Wailing Dunes 182 1 Artist's Shack (Gravesite Plain) 1 1 Royal Grave Evergaol 151 1 Suppressing Pillar 17 1

**The bottom 20 locations in Elden Ring, in terms of unique items, show that many areas have very few or no items at all. Locations like Castle Ensis, Belurat Tower Settlement, Midra's Manse, Enir-Ilim, and Shadow Keep have 0 unique items, while others such as Sellia Evergaol, Malefactor's Evergaol, and Golden Lineage Evergaol have only 1 unique item. This suggests that smaller dungeons, minor towers, or less-explored zones generally offer minimal loot compared to major castles and academies.**

# **📊 9: Load location-NPCs bridge table**


In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_relationships 
FROM eldenringcatalog.gold.bridge_location_npcs;

total_relationships 215

In [0]:
%sql
SELECT DISTINCT * 
FROM eldenringcatalog.gold.bridge_location_npcs 
LIMIT 20;

location_npc_key location_id npc_name created_at 0 0 Spirit NPC 2025-11-29T17:01:19.511Z 1 3 Moore 2025-11-29T17:01:19.511Z 2 3 Sir Ansbach 2025-11-29T17:01:19.511Z 3 3 Hornsent Grandam 2025-11-29T17:01:19.511Z 4 4 Rellana 2025-11-29T17:01:19.511Z 5 4 Twin Moon Knight 2025-11-29T17:01:19.511Z 6 8 Spirit NPC 2025-11-29T17:01:19.511Z 7 14 Spirit NPC 2025-11-29T17:01:19.511Z 8 16 Spirit NPC 2025-11-29T17:01:19.511Z 9 21 Count Ymir 2025-11-29T17:01:19.511Z 10 21 Jolan 2025-11-29T17:01:19.511Z 11 22 Fire Knight Queelign 2025-11-29T17:01:19.511Z 12 25 Moore 2025-11-29T17:01:19.511Z 13 25 Thiollier 2025-11-29T17:01:19.511Z 14 25 Sir Ansbach 2025-11-29T17:01:19.511Z 15 25 Dryleaf Dane 2025-11-29T17:01:19.511Z 16 25 Redmane Freyja 2025-11-29T17:01:19.511Z 17 25 Needle Knight Leda 2025-11-29T17:01:19.511Z 18 26 Swordhand of Night Anna 2025-11-29T17:01:19.511Z 19 29 Dryleaf Dane 2025-11-29T17:01:19.511Z

**The bridge_location_npcs table contains 215 unique relationships between locations and NPCs in Elden Ring. The sample shows that some locations host multiple NPCs, like location 3 with characters such as Moore, Sir Ansbach, and Hornsent Grandam, while other locations may only host a single NPC or a generic Spirit NPC. This table effectively maps which NPCs appear in which locations, helping to track quests, summons, and interactions across the game world.**

# **📊 10: Which locations have the most NPCs?**

In [0]:
%sql
SELECT 
    l.location_name,
    l.location_id,
    COUNT(DISTINCT*) AS npc_count
FROM eldenringcatalog.gold.bridge_location_npcs b
JOIN eldenringcatalog.gold.dim_locations l ON b.location_id = l.location_id
GROUP BY l.location_name, l.location_id
ORDER BY npc_count DESC
LIMIT 20;

location_name location_id npc_count Roundtable Hold 52 14 Seethewater Cave 215 13 Bestial Sanctum 232 11 Sealed Tunnel 225 7 Raya Lucaria Academy 144 6 Volcano Manor 217 5 Leyndell, Royal Capital 230 5 Stormveil Castle 81 4 Ranni's Rise 143 4 Deeproot Depths 272 3 Stone Coffin Fissure 41 3 Stormfoot Catacombs 76 3 Stormhill 78 3 Siofra River 283 3 Leyndell, Ashen Capital 229 3 Mistwood 70 3 Redmane Castle 173 3 Gowry's Shack 170 2 Bellum Church 111 2 Academy Gate Town 108 2

**The top 20 locations with the most NPCs show that Roundtable Hold leads with 14 NPCs, followed by Seethewater Cave with 13 and Bestial Sanctum with 11. Many major areas like Raya Lucaria Academy, Leyndell, Royal Capital, and Volcano Manor host 5–6 NPCs each, indicating these locations are key hubs for interactions, quests, and gameplay events. Smaller locations tend to host 3–4 NPCs, reflecting a mix of major quest areas and minor exploration zones. This distribution highlights the concentration of NPCs in central hubs versus spread-out adventuring areas.**

# **📊 11: Load location-bosses bridge table**

In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_relationships 
FROM eldenringcatalog.gold.bridge_location_bosses

total_relationships 235

In [0]:
%sql
SELECT DISTINCT * 
FROM eldenringcatalog.gold.bridge_location_bosses 
LIMIT 20;

location_boss_key location_id boss_name created_at 0 2 Demi-Human Swordmaster Onze 2025-11-29T17:01:24.609Z 1 3 Divine Beast Dancing Lion 2025-11-29T17:01:24.609Z 2 4 Rellanas Cameo 2025-11-29T17:01:24.609Z 3 4 Spelldrake Talisman 2025-11-29T17:01:24.609Z 4 7 Ancient Dragon-Man 2025-11-29T17:01:24.609Z 5 10 Black Knight Garrew 2025-11-29T17:01:24.609Z 6 11 Red Bear 2025-11-29T17:01:24.609Z 7 13 Chief Bloodfiend 2025-11-29T17:01:24.609Z 8 18 Blackgaol Knight 2025-11-29T17:01:24.609Z 9 19 Curseblade Labirith 2025-11-29T17:01:24.609Z 10 21 Count Ymir 2025-11-29T17:01:24.609Z 11 21 Mother of Fingers 2025-11-29T17:01:24.609Z 12 21 Swordhand of Night Anna 2025-11-29T17:01:24.609Z 13 21 Swordhand of Night Jolan 2025-11-29T17:01:24.609Z 14 22 Fire Knight Queelign 2025-11-29T17:01:24.609Z 15 23 Jori 2025-11-29T17:01:24.609Z 16 23 Elder Inquisitor 2025-11-29T17:01:24.609Z 17 24 Rakshasa 2025-11-29T17:01:24.609Z 18 25 Promised Consort Radahn 2025-11-29T17:01:24.609Z 19 25 Radahn 2025-11-29T17:01:24.609Z

**The bridge_location_bosses table contains 235 unique boss-location relationships, showing which bosses appear in which locations. The sample of 20 rows highlights that some locations host multiple bosses, such as location_id 21, which contains Count Ymir, Mother of Fingers, and Swordhand of Night Anna and Jolan. Other locations host a single boss, like Demi-Human Swordmaster Onze at location 2. This indicates that boss encounters are unevenly distributed, with certain locations serving as major challenge hubs.**

# **📊 12: Which locations have multiple bosses?**

In [0]:
%sql
SELECT 
    l.location_name,
    l.location_id,
    COUNT(DISTINCT *) AS boss_count
FROM eldenringcatalog.gold.bridge_location_bosses b
JOIN eldenringcatalog.gold.dim_locations l ON b.location_id = l.location_id
GROUP BY l.location_name, l.location_id
HAVING COUNT(*) > 1
ORDER BY boss_count DESC;

location_name location_id boss_count Leyndell, Royal Capital 230 6 Leyndell, Ashen Capital 229 6 Volcano Manor 217 6 Elden Throne 226 4 Stormveil Castle 81 4 Ainsel River 270 4 Cathedral of Manus Metyr 21 4 Mistwood 70 3 Forbidden Lands 246 3 Stormhill 78 3 Lake of Rot 276 3 Moonlight Altar 141 3 Nokron, Eternal City 280 3 Raya Lucaria Academy 144 3 Redmane Castle 173 3 Yelough Anix Tunnel 269 2 Altus Tunnel 184 2 Grand Cloister 274 2 Perfumer's Grotto 194 2 Mohgwyn Palace 277 2 Bestial Sanctum 232 2 Miquella's Haligtree 266 2 Fringefolk Hero's Grave 64 2 Malefactor's Evergaol 137 2 Hallowhorn Grounds 275 2 Darklight Catacombs 23 2 Cuckoo's Evergaol 125 2 Black Knife Catacombs 113 2 Sage's Cave 196 2 Roundtable Hold 52 2 Mohgwyn Dynasty Mausoleum 278 2 Auriza Hero's Grave 218 2 Siofra River 283 2 Leyndell Catacombs 227 2 Ringleader's Evergaol 148 2 Elphael Brace of the Haligtree 263 2 Finger Ruins of Miyr 26 2 Castle Ensis 4 0 Midra's Manse 46 0 Enir-Ilim 25 0

**Several locations in Elden Ring host multiple bosses, making them key challenge areas. Leyndell, Royal Capital, Leyndell, Ashen Capital, and Volcano Manor top the list with 6 bosses each, followed by Elden Throne, Stormveil Castle, and Ainsel River with 4 bosses. Other locations like Mistwood, Raya Lucaria Academy, and Redmane Castle have 3 bosses, while many smaller or optional areas host 2. This distribution shows that some major hubs concentrate multiple boss encounters, whereas other locations may only feature a single or no boss.**

# **📊 13: Load location-creatures bridge table**

In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_relationships
FROM eldenringcatalog.gold.bridge_location_creatures;

total_relationships 789

In [0]:
%sql
SELECT DISTINCT *
FROM eldenringcatalog.gold.bridge_location_creatures
LIMIT 20;

location_creature_key location_id creature_name created_at 0 0 Man-Fly 2025-11-29T17:01:22.030Z 1 2 Living Jar 2025-11-29T17:01:22.030Z 2 2 Jar Innards 2025-11-29T17:01:22.030Z 3 2 Shadow Undead 2025-11-29T17:01:22.030Z 4 2 Messmer Foot Soldier 2025-11-29T17:01:22.030Z 5 3 Man-Fly 2025-11-29T17:01:22.030Z 6 3 Gravebird 2025-11-29T17:01:22.030Z 7 3 Curseblade 2025-11-29T17:01:22.030Z 8 3 Horned Warrior 2025-11-29T17:01:22.030Z 9 3 Spider Scorpion 2025-11-29T17:01:22.030Z 10 3 Shadow Undead 2025-11-29T17:01:22.030Z 11 3 Fire Knight Queelign 2025-11-29T17:01:22.030Z 12 3 Lesser Ulcerated Tree Spirit 2025-11-29T17:01:22.030Z 13 4 Page 2025-11-29T17:01:22.030Z 14 4 Eagle 2025-11-29T17:01:22.030Z 15 4 Turtle 2025-11-29T17:01:22.030Z 16 4 Rotten Stray 2025-11-29T17:01:22.030Z 17 4 Troll Knight 2025-11-29T17:01:22.030Z 18 4 Messmer Soldier 2025-11-29T17:01:22.030Z 19 4 Noble Sorcerer 2025-11-29T17:01:22.030Z

**The bridge_location_creatures table in Elden Ring contains 789 unique location-creature relationships, mapping various creatures to the locations where they appear. A sample of 20 entries shows a wide variety of creatures, from "Man-Fly" and "Living Jar" to "Troll Knight" and "Noble Sorcerer," highlighting the diversity of enemies and entities populating the game world.**

# **📊 14: Which locations have the most creature variety?**

In [0]:
%sql
SELECT 
    l.location_name,
    lc.location_id,
    COUNT(DISTINCT *) AS creature_count,
    COUNT(DISTINCT lc.creature_name) AS unique_creatures
FROM eldenringcatalog.gold.bridge_location_creatures lc
LEFT JOIN eldenringcatalog.gold.dim_locations l
    ON lc.location_id = l.location_id
GROUP BY l.location_name, lc.location_id
ORDER BY unique_creatures DESC
LIMIT 20;

location_name location_id creature_count unique_creatures Raya Lucaria Academy 144 22 22 Shadow Keep 33 0 14 Stormhill 78 14 14 Mistwood 70 14 14 Siofra River 283 13 13 The Shaded Castle 200 12 12 Castle Ensis 4 0 12 Moonlight Altar 141 11 11 Miquella's Haligtree 266 11 11 Volcano Manor 217 11 11 Swamp of Aeonia 181 10 10 Caria Manor 115 10 10 Dragon-Burnt Ruins 61 9 9 Enir-Ilim 25 0 9 Castle Morne 90 8 8 Belurat Tower Settlement 3 0 8 Stormveil Castle 81 8 8 Elphael Brace of the Haligtree 263 8 8 Minor Erdtree (Weeping Peninsula) 98 7 7 Nokstella, Eternal City 281 7 7

**The analysis of location-creature relationships in Elden Ring shows that Raya Lucaria Academy hosts the most diverse set of creatures with 22 unique types, followed by locations like Shadow Keep, Stormhill, and Mistwood, each featuring 14 distinct creatures. Several other areas, including Siofra River, The Shaded Castle, and Miquella's Haligtree, contain 10–13 unique creatures. Interestingly, some locations such as Castle Ensis, Enir-Ilim, and Belurat Tower Settlement have zero total creature entries despite still listing unique creature types. Overall, this highlights a mix of highly populated areas with rich creature diversity and more sparse locations across the Lands Between.**

# **📊 15: What are the most common creatures**?


In [0]:
%sql
SELECT 
    creature_name,
    COUNT(DISTINCT *) AS location_count
FROM eldenringcatalog.gold.bridge_location_creatures
GROUP BY creature_name
ORDER BY location_count DESC
LIMIT 30;

creature_name location_count Map Link 24 Rotten Stray 21 Giant Rat 20 Miranda Sprout 14 Putrid Corpse 13 Giant Miranda Sprout 12 Wandering Nobles 12 Giant Bat 12 Teardrop Scarab 11 Rat 11 Soldier 11 Imp 10 Skeletal Slime 9 Demi-Human 9 Living Jar 9 Turtle 8 Giant Crab 8 Marionette Soldier 8 Fanged Imp 8 Shadow Undead 8 Misbegotten 8 Wolf 8 Sheep 7 Deer 7 Troll 7 Godrick Soldier 7 Godrick Knight 7 Messmer Soldier 7 Miner 7 Servant of Rot 7

**The creature distribution shows that several enemies appear across multiple locations in Elden Ring. Excluding the data issue with “Map Link,” the most common creatures are Rotten Stray (21 locations), Giant Rat (20), Miranda Sprout (14), and Putrid Corpse (13). Many other enemies, like Giant Miranda Sprout, Wandering Nobles, and Giant Bat, appear in 12 locations, indicating that certain enemy types are recurrent across the Lands Between, providing consistent challenges for players.**

# **📊 16: Rare creatures (appearing in only 1-2 locations**)

In [0]:
%sql
SELECT 
    creature_name,
    COUNT(DISTINCT *) AS location_count
FROM eldenringcatalog.gold.bridge_location_creatures
GROUP BY creature_name
HAVING COUNT( *) <= 2
ORDER BY location_count ASC, creature_name ASC;


creature_name location_count Ash of War: Flame Skewer 1 Blistered Stray 1 Bloodhound Knight 1 Blue Worm 1 Burning Slug 1 Cat Imp 1 Cerulean Scarabs 1 Colossal Fingercreeper 1 Colossal Fingercreepers 1 Crimson Scarabs 1 Crimson Teardrop Scarab 1 Crystalians 1 Demi-Human Queen 1 Disciple of Okina 1 Divine Bird Warrior 1 Eagles 1 Edgar the Revenger 1 Eleonora 1 Festering Fingerprint Vyke 1 Festering Fingerprint Vyke (Invader) 1 Finger Mimic 1 Fingercreepers 1 Fire Knight 1 Fire Knight Captain Kood 1 Fire Knight Queelign 1 First-Generation Albinauric 1 Flame Guardian 1 Furnace Golem 1 Giant Beast Skeleton 1 Giant Dog 1 Giant Land Squirt 1 Giant Oracle Envoy 1 Giant Skeletal Slime 1 Glintstone Dragon 1 Godrick Foot Soldier 1 Grave Warden Duelist 1 Greater Potentate 1 Grossmesser 1 Haligtree Knight 1 Haligtree Soldier 1 High Page 1 Highwayman 1 Hippopotamus 1 Hornsent 1 Inaba 1 Inquisitor Ghiza 1 Knight of Zamor 1 Large Oracle Envoy 1 Lesser Black Knife Assassin 1 Lesser Black Knight 1 Lesser Cemetery Shade 1 Lesser Crystalian 1 Lesser Fingercreepers 1 Lesser Magma Wyrm 1 Lesser Sanguine Noble 1 Lesser Wormface 1 Lion Guardian 1 Living Jars 1 Mage-Serpent 1 Magnus the Beast Claw 1 Mausoleum Knight 1 Mausoleum Soldier 1 Moongrum 1 Moonrithyll 1 Nameless White Mask 1 Necromancer 1 Nomad 1 Nox Monk 1 Onyx Lord 1 Oracle Envoy 1 Owl 1 Poison Flowers 1 Putrid Avatar 1 Putrid Tree Spirit 1 Rats 1 Raya Lucaria Foot Soldier 1 Raya Lucaria Knight 1 Raya Lucaria Soldiers 1 Shadow Militia 1 Shadow Militiamen 1 Shadowpot 1 Skeletal Militiaman 1 Skeletal Militiamen 1 Soldier of Godrick 1 Starcaller 1 Thorn Sorcerer 1 Troll Spirit Knight 1 Valiant Gargoyle 1 Violet Bloody Finger 1 Vulgar Militia 1 Wandering Mausoleum 1 Wild Mouflon 1 Alabaster Lord 2 Albinauric Wolfback Archer 2 Anastasia 2 Blackflame Monk 2 Bloody Finger Okina 2 Boar 2 Carian Knight 2 Celebrant 2 Crystalian 2 Depraved Perfumer 2 Direwolf 2 Dog 2 Fallen Hawks Soldier 2 Giant Ant 2 Giant Poison Flower 2 Guillemot 2 Horned Shaman 2 Lesser Alabaster Lord 2 Lesser Bloodhound Knight 2 Lesser Crucible Knight 2 Leyndell Soldier 2 Lookout Stones 2 Man-Serpent 2 Perfumer 2 Raya Lucaria Soldier 2 Sanguine Noble 2 Serpent Snail 2 Skeletal Bandit 2 Skeleton Mage 2 Slug 2 Spider Scorpion 2 Tarnished-Eater 2 Troll Knight 2 Ulcerated Tree Spirit 2 Warhawk 2

# **📊 17: Item type distribution per location**

In [0]:
%sql
SELECT 
    l.location_name,
    i.item_type,
    COUNT(DISTINCT *) AS item_count
FROM eldenringcatalog.gold.bridge_location_items bli
INNER JOIN eldenringcatalog.gold.dim_locations l ON bli.location_id = l.location_id
LEFT JOIN eldenringcatalog.gold.dim_items i ON bli.item_name = i.item_name
GROUP BY l.location_name, i.item_type
ORDER BY l.location_name, item_count DESC
LIMIT 50;

location_name item_type item_count Abandoned Ailing Village Material 3 Abandoned Ailing Village Upgrade Material 1 Abandoned Ailing Village null 0 Abandoned Ailing Village Key Item 0 Abandoned Cave Consumable 3 Abandoned Cave Material 2 Abandoned Cave null 0 Abandoned Church Upgrade Material 0 Abandoned Church null 0 Academy Crystal Cave Material 4 Academy Crystal Cave Consumable 2 Academy Crystal Cave null 0 Academy Crystal Cave Key Item 0 Academy Gate Town Material 2 Academy Gate Town Consumable 2 Academy Gate Town Upgrade Material 1 Academy Gate Town null 0 Ailing Village Material 5 Ailing Village Upgrade Material 1 Ailing Village Consumable 1 Ailing Village null 0 Ainsel River Material 4 Ainsel River Upgrade Material 1 Ainsel River null 0 Ainsel River Key Item 0 Ainsel River Main Consumable 2 Ainsel River Main Material 1 Ainsel River Main Key Item 0 Ainsel River Main null 0 Albinauric Rise null 0 Albinauric's Shack Material 1 Albinauric's Shack null 0 Altus Tunnel Material 2 Altus Tunnel Consumable 2 Altus Tunnel null 0 Apostate Derelict null 0 Artist's Shack (Gravesite Plain) null 0 Artist's Shack (Limgrave) Material 4 Artist's Shack (Limgrave) null 0 Artist's Shack (Liurnia of the Lakes) null 0 Auriza Hero's Grave Material 1 Auriza Hero's Grave Ammo 0 Auriza Hero's Grave Key Item 0 Auriza Hero's Grave null 0 Auriza Side Tomb Material 1 Auriza Side Tomb null 0 Auriza Side Tomb Key Item 0 Bellum Church Material 1 Bellum Church Upgrade Material 1 Bellum Church Consumable 1

# **📊 18: Locations with most "Ammo" items**

In [0]:
%sql
SELECT 
    l.location_name,
    COUNT(*) AS ammo_count
FROM eldenringcatalog.gold.bridge_location_items bli
INNER JOIN eldenringcatalog.gold.dim_items i ON bli.item_name = i.item_name
INNER JOIN eldenringcatalog.gold.dim_locations l ON bli.location_id = l.location_id
WHERE i.item_type = 'Ammo'
GROUP BY l.location_name
ORDER BY ammo_count DESC
LIMIT 20;

location_name ammo_count Hermit Merchant's Shack 7 Isolated Merchant's Shack Weeping Peninsula 4 Stormveil Castle 3 Isolated Merchant's Shack (Dragonbarrow) 3 Castle Morne 2 Elphael Brace of the Haligtree 2 Bridge of Sacrifice 2 Castle Ensis 2 Belurat Tower Settlement 2 Enir-Ilim 2 Miquella's Haligtree 1 Volcano Manor 1 Tower of Return 1 Bonny Gaol 1 Ellac Greatbridge 1 Stormcaller Church 1 Subterranean Shunning-Grounds 1 Deeproot Depths 1 Fog Rift Fort 1 Stillwater Cave 1

# **📊 19: Create location "density" score**

In [0]:
%sql
WITH location_stats AS (
    SELECT 
        l.location_id,
        l.location_name,
        COALESCE(COUNT(DISTINCT bli.item_name), 0) AS item_count,
        COALESCE(COUNT(DISTINCT bln.npc_name), 0) AS npcs,
        COALESCE(COUNT(DISTINCT blb.boss_name), 0) AS bosses,
        COALESCE(COUNT(DISTINCT blc.creature_name), 0) AS creatures
    FROM eldenringcatalog.gold.dim_locations l
    LEFT JOIN eldenringcatalog.gold.bridge_location_items bli ON l.location_id = bli.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_npcs bln ON l.location_id = bln.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_bosses blb ON l.location_id = blb.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_creatures blc ON l.location_id = blc.location_id
    GROUP BY l.location_id, l.location_name
)
SELECT 
    location_name,
    item_count,
    npcs,
    bosses,
    creatures,
    ROUND(item_count + (npcs * 5) + (bosses * 10) + (creatures * 0.5), 2) AS density_score
FROM location_stats
ORDER BY density_score DESC
LIMIT 30;

location_name item_count npcs bosses creatures density_score Leyndell, Royal Capital 75 5 6 1 160.5 Stormveil Castle 82 4 4 8 146.0 Volcano Manor 52 5 6 11 142.5 Raya Lucaria Academy 63 6 3 22 134.0 Shadow Keep 93 4 1 14 130.0 Enir-Ilim 53 6 3 9 117.5 Roundtable Hold 10 14 2 0 100.0 Seethewater Cave 18 13 1 6 96.0 Leyndell, Ashen Capital 14 3 6 3 90.5 Bestial Sanctum 12 11 2 0 87.0 Castle Ensis 47 2 2 12 83.0 Belurat Tower Settlement 52 3 1 8 81.0 Mistwood 27 3 3 14 79.0 Elphael Brace of the Haligtree 49 1 2 8 78.0 Siofra River 36 3 2 13 77.5 Nokron, Eternal City 39 1 3 7 77.5 Ainsel River 29 1 4 1 74.5 Redmane Castle 19 3 3 6 67.0 Stormhill 14 3 3 14 66.0 Cathedral of Manus Metyr 14 2 4 3 65.5 Mohgwyn Palace 28 2 2 6 61.0 Sealed Tunnel 13 7 1 4 60.0 Lake of Rot 26 0 3 5 58.5 Miquella's Haligtree 32 0 2 11 57.5 Moonlight Altar 16 1 3 11 56.5 Deeproot Depths 31 3 1 1 56.5 The Shaded Castle 27 2 1 12 53.0 Subterranean Shunning-Grounds 41 2 0 0 51.0 Caria Manor 24 2 1 10 49.0 Elden Throne 3 1 4 0 48.0

# **📊 20: Merchant NPCs and their locations**

In [0]:
%sql
SELECT npc_name, role, is_dlc
FROM eldenringcatalog.gold.dim_npcs
WHERE role LIKE '%Merchant%' OR role LIKE '%Shop%';

npc_name role is_dlc Abandoned Merchant Siofra River General Goods Shop 0 Brother Corhyn Merchant and Quest NPC 0 Count Ymir, High Priest Merchant 1 Gatekeeper Gostoc NPC and Merchant 0 Hermit Merchant (Leyndell) Merchant 0 Imprisoned Merchant General Goods Shop 0 Isolated Merchant Raya Lucaria General Goods Shop 0 Knight Bernahl Ash of War Merchant 0 Merchant Kale General Goods Shop 0 Miriel, Pastor of Vows Merchant 0 Moore Merchant 1 Nomadic Merchant Central ALTUS PLATEAU General Goods Shop 0 Patches Boss, Merchant, Quest NPC, Summon 0 Pidia, Carian Servant Merchant 0 Preceptor Seluvis Quest NPC and Merchant 0 Sorcerer Rogier Merchant, Quest NPC, and Summon 0 Sorceress Sellen Merchant and Quest NPC 0 Thiollier Merchant, NPC Quest Related 1 Twin Maiden Husks Merchant 0

In [0]:
%sql
SELECT 
    n.npc_name,
    l.location_name,
    n.role
FROM eldenringcatalog.gold.dim_npcs n
LEFT JOIN eldenringcatalog.gold.bridge_location_npcs bln ON n.npc_name = bln.npc_name
LEFT JOIN eldenringcatalog.gold.dim_locations l ON bln.location_id = l.location_id
WHERE n.role LIKE '%Merchant%' OR n.role LIKE '%Shop%'

npc_name location_name role Abandoned Merchant Siofra River null General Goods Shop Brother Corhyn Leyndell, Royal Capital Merchant and Quest NPC Count Ymir, High Priest null Merchant Gatekeeper Gostoc Stormveil Castle NPC and Merchant Hermit Merchant (Leyndell) null Merchant Imprisoned Merchant Mohgwyn Dynasty Mausoleum General Goods Shop Isolated Merchant Raya Lucaria null General Goods Shop Knight Bernahl Volcano Manor Ash of War Merchant Merchant Kale null General Goods Shop Miriel, Pastor of Vows null Merchant Moore Enir-Ilim Merchant Nomadic Merchant Central ALTUS PLATEAU null General Goods Shop Patches Volcano Manor Boss, Merchant, Quest NPC, Summon Pidia, Carian Servant null Merchant Preceptor Seluvis Seluvis's Rise Quest NPC and Merchant Sorcerer Rogier Stormveil Castle Merchant, Quest NPC, and Summon Sorceress Sellen Raya Lucaria Academy Merchant and Quest NPC Thiollier Stone Coffin Fissure Merchant, NPC Quest Related Twin Maiden Husks Roundtable Hold Merchant Brother Corhyn Leyndell, Ashen Capital Merchant and Quest NPC Knight Bernahl Warmaster's Shack Ash of War Merchant Moore Belurat Tower Settlement Merchant Patches The Shaded Castle Boss, Merchant, Quest NPC, Summon Preceptor Seluvis Ranni's Rise Quest NPC and Merchant Sorcerer Rogier Roundtable Hold Merchant, Quest NPC, and Summon Sorceress Sellen Witchbane Ruins Merchant and Quest NPC Thiollier Enir-Ilim Merchant, NPC Quest Related Knight Bernahl Stormhill Ash of War Merchant Patches Murkwater Cave Boss, Merchant, Quest NPC, Summon Sorceress Sellen Waypoint Ruins Merchant and Quest NPC

# **📊 21: Quest NPCs and their locations**

In [0]:
%sql
SELECT npc_name, role, is_dlc
FROM eldenringcatalog.gold.dim_npcs
WHERE role LIKE '%Quest%';

npc_name role is_dlc Sir Ansbach Pureblood Knight Captain (former), NPC Quest Related 1 Blaidd the Half-Wolf Quest NPC and summon 0 Bloody Finger Hunter Yura Quest NPC and Summonable Cooperator 0 Boc the Seamster Garments Adjuster andQuest NPC 0 Brother Corhyn Merchant and Quest NPC 0 D, Beholder of Death Quest NPC and Summon 0 D, Hunter of the Dead Quest NPC and Summon 0 Knight Diallos Quest NPC 0 Dragon Communion Priestess Questline Related 1 Dung Eater Quest NPC, Invader, and Summon 0 Edgar Morne Castellan, Quest NPC 0 Fia, Deathbed Companion Quest NPC 0 Redmane Freyja Cooperator, Questline Related 1 Goldmask Quest NPC 0 Hyetta Quest NPC 0 Igon Drake Warrior, NPC Quest Related 1 Irina Quest NPC 0 Jar Bairn Quest NPC 0 Latenna Quest NPC 0 Millicent Quest NPC and Summonable Cooperator 0 Nepheli Loux, Warrior Quest NPC and Summon 0 Patches Boss, Merchant, Quest NPC, Summon 0 Preceptor Seluvis Quest NPC and Merchant 0 Sorcerer Rogier Merchant, Quest NPC, and Summon 0 Sorceress Sellen Merchant and Quest NPC 0 St. Trina NPC Quest Related 1 Tanith Quest NPC 0 Thiollier Merchant, NPC Quest Related 1 Three Fingers Quest NPC 0 Wandering Artist Spirit Quest NPC 0 Iji War Counselor & Carian Blacksmith Blacksmith & Quest NPC 0 White Mask Varre Guide and Quest NPC 0 Witch-Hunter Jerren Quest NPC and Summon 0

In [0]:
%sql
SELECT 
    n.npc_name,
    l.location_name,
    n.role
FROM eldenringcatalog.gold.dim_npcs n
LEFT JOIN eldenringcatalog.gold.bridge_location_npcs bln ON n.npc_name = bln.npc_name
LEFT JOIN eldenringcatalog.gold.dim_locations l ON bln.location_id = l.location_id
WHERE n.role LIKE '%Quest%'

npc_name location_name role Sir Ansbach Enir-Ilim Pureblood Knight Captain (former), NPC Quest Related Blaidd the Half-Wolf null Quest NPC and summon Bloody Finger Hunter Yura Second Church of Marika Quest NPC and Summonable Cooperator Boc the Seamster Leyndell, Royal Capital Garments Adjuster andQuest NPC Brother Corhyn Leyndell, Royal Capital Merchant and Quest NPC D, Beholder of Death null Quest NPC and Summon D, Hunter of the Dead null Quest NPC and Summon Knight Diallos null Quest NPC Dragon Communion Priestess Grand Altar of Dragon Communion Questline Related Dung Eater Subterranean Shunning-Grounds Quest NPC, Invader, and Summon Edgar Castle Morne Morne Castellan, Quest NPC Fia, Deathbed Companion null Quest NPC Redmane Freyja Shadow Keep Cooperator, Questline Related Goldmask Leyndell, Royal Capital Quest NPC Hyetta Frenzied Flame Proscription Quest NPC Igon null Drake Warrior, NPC Quest Related Irina Bridge of Sacrifice Quest NPC Jar Bairn Jarburg Quest NPC Latenna Slumbering Wolf's Shack Quest NPC Millicent Elphael Brace of the Haligtree Quest NPC and Summonable Cooperator Nepheli Loux, Warrior null Quest NPC and Summon Patches Volcano Manor Boss, Merchant, Quest NPC, Summon Preceptor Seluvis Seluvis's Rise Quest NPC and Merchant Sorcerer Rogier Stormveil Castle Merchant, Quest NPC, and Summon Sorceress Sellen Raya Lucaria Academy Merchant and Quest NPC St. Trina Stone Coffin Fissure NPC Quest Related Tanith Volcano Manor Quest NPC Thiollier Stone Coffin Fissure Merchant, NPC Quest Related Three Fingers Frenzied Flame Proscription Quest NPC Wandering Artist Spirit null Quest NPC Iji War Counselor & Carian Blacksmith null Blacksmith & Quest NPC White Mask Varre Rose Church Guide and Quest NPC Witch-Hunter Jerren Redmane Castle Quest NPC and Summon Sir Ansbach Belurat Tower Settlement Pureblood Knight Captain (former), NPC Quest Related Bloody Finger Hunter Yura Raya Lucaria Academy Quest NPC and Summonable Cooperator Brother Corhyn Leyndell, Ashen Capital Merchant and Quest NPC Dung Eater Leyndell, Royal Capital Quest NPC, Invader, and Summon Redmane Freyja Enir-Ilim Cooperator, Questline Related Goldmask Leyndell, Ashen Capital Quest NPC Hyetta Purified Ruins Quest NPC Irina Bridge of Sacrifice Quest NPC Millicent Dominula, Windmill Village Quest NPC and Summonable Cooperator Patches The Shaded Castle Boss, Merchant, Quest NPC, Summon Preceptor Seluvis Ranni's Rise Quest NPC and Merchant Sorcerer Rogier Roundtable Hold Merchant, Quest NPC, and Summon Sorceress Sellen Witchbane Ruins Merchant and Quest NPC Thiollier Enir-Ilim Merchant, NPC Quest Related White Mask Varre Church of Inhibition Guide and Quest NPC Witch-Hunter Jerren Raya Lucaria Academy Quest NPC and Summon Dung Eater Leyndell, Ashen Capital Quest NPC, Invader, and Summon Hyetta Bellum Highway Quest NPC Millicent Swamp of Aeonia Quest NPC and Summonable Cooperator Patches Murkwater Cave Boss, Merchant, Quest NPC, Summon Sorceress Sellen Waypoint Ruins Merchant and Quest NPC Witch-Hunter Jerren Witchbane Ruins Quest NPC and Summon Dung Eater Roundtable Hold Quest NPC, Invader, and Summon Hyetta Bellum Church Quest NPC Millicent Street of Sages Ruins Quest NPC and Summonable Cooperator Millicent Gowry's Shack Quest NPC and Summonable Cooperator Millicent Church of the Plague Quest NPC and Summonable Cooperator

# **📊 22: Summonable NPCs**

In [0]:
%sql
SELECT npc_name, role, is_dlc
FROM eldenringcatalog.gold.dim_npcs
WHERE role LIKE '%Summon%';

npc_name role is_dlc Bloody Finger Hunter Yura Quest NPC and Summonable Cooperator 0 D, Beholder of Death Quest NPC and Summon 0 D, Hunter of the Dead Quest NPC and Summon 0 Dung Eater Quest NPC, Invader, and Summon 0 Finger Maiden Therolina NPC Summon 0 Lionel the Lionhearted Summon and Boss 0 Millicent Quest NPC and Summonable Cooperator 0 Nepheli Loux, Warrior Quest NPC and Summon 0 Patches Boss, Merchant, Quest NPC, Summon 0 Shabriri NPC and Summonable Cooperator 0 Sorcerer Rogier Merchant, Quest NPC, and Summon 0 Witch-Hunter Jerren Quest NPC and Summon 0

# **📊23: Boss-only locations (locations with bosses but few other entities)**

In [0]:
%sql
WITH location_stats AS (
    SELECT 
        l.location_id,
        l.location_name,
        COALESCE(COUNT(DISTINCT bli.item_name), 0) AS item_count,
        COALESCE(COUNT(DISTINCT bln.npc_name), 0) AS npcs,
        COALESCE(COUNT(DISTINCT blb.boss_name), 0) AS bosses,
        COALESCE(COUNT(DISTINCT blc.creature_name), 0) AS creatures
    FROM eldenringcatalog.gold.dim_locations l
    LEFT JOIN eldenringcatalog.gold.bridge_location_items bli ON l.location_id = bli.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_npcs bln ON l.location_id = bln.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_bosses blb ON l.location_id = blb.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_creatures blc ON l.location_id = blc.location_id
    GROUP BY l.location_id, l.location_name
)
SELECT location_name, bosses, item_count, creatures
FROM location_stats
WHERE bosses >= 1 AND item_count <= 5 AND npcs = 0;

location_name bosses item_count creatures Minor Erdtree (Caelid) 1 4 1 Ringleader's Evergaol 2 3 5 Eastern Nameless Mausoleum 1 5 0 Fort Faroth 1 5 3 Minor Erdtree (Mt. Gelmir) 1 0 0 Weeping Evergaol 1 2 0 Cuckoo's Evergaol 2 1 1 Caelem Ruins 1 2 2 Lord Contender's Evergaol 1 2 1 Cathedral of Dragon Communion (Caelid) 1 4 2 Minor Erdtree (Altus Plateau) 1 4 1 Divine Tower of East Altus 1 3 0 Malefactor's Evergaol 2 1 0 Royal Grave Evergaol 1 1 0 Shaman Village 1 2 0 Foot of the Jagged Peak 1 2 0 Minor Erdtree (Mountaintops East) 1 3 2 Minor Erdtree (Dragonbarrow) 1 2 0 Writheblood Ruins 1 3 4 Western Nameless Mausoleum 1 5 1 Northern Nameless Mausoleum 1 5 0 Stormhill Evergaol 1 2 0 Sellia Evergaol 1 1 0

# **📊 24: Safe locations (no bosses, no enemies)**

In [0]:
%sql
WITH location_stats AS (
    SELECT 
        l.location_id,
        l.location_name,
        COALESCE(COUNT(DISTINCT bli.item_name), 0) AS item_count,
        COALESCE(COUNT(DISTINCT bln.npc_name), 0) AS npcs,
        COALESCE(COUNT(DISTINCT blb.boss_name), 0) AS bosses,
        COALESCE(COUNT(DISTINCT blc.creature_name), 0) AS creatures
    FROM eldenringcatalog.gold.dim_locations l
    LEFT JOIN eldenringcatalog.gold.bridge_location_items bli ON l.location_id = bli.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_npcs bln ON l.location_id = bln.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_bosses blb ON l.location_id = blb.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_creatures blc ON l.location_id = blc.location_id
    GROUP BY l.location_id, l.location_name
)
SELECT location_name, item_count, npcs
FROM location_stats
WHERE bosses = 0 AND creatures = 0;

location_name item_count npcs Slumbering Wolf's Shack 2 1 Gowry's Shack 6 2 Craftsman's Shack 1 0 Artist's Shack (Gravesite Plain) 1 0 Isolated Merchant's Shack Weeping Peninsula 12 1 Subterranean Shunning-Grounds 41 2 Church of Benediction 7 0 Boilprawn Shack 3 1 Deep Siofra Well 0 0 Run-Down Traveler's Rest 2 0 Grand Altar of Dragon Communion 4 1 Oridys's Rise 1 0 Whipping Hut 1 1 Ranni's Rise 1 4 Scadutree Chalice 3 0 Southern Nameless Mausoleum 9 0 Grand Lift of Dectus 2 0 Grand Lift of Rold 3 1 Deep Ainsel Well 0 0 Rabbath's Rise 3 0 Stormhill Shack 4 1 Starfall Crater 0 0 Suppressing Pillar 1 0 Grand Lift of Dectus 2 0 Minor Erdtree Church 3 0 Leyndell Colosseum 0 0 Converted Tower 3 0 Divine Tower of West Altus 1 0 Renna's Rise 1 0

# **📊 25: Locations with unique items (appearing only once)**

In [0]:
%sql
WITH item_counts AS (
    SELECT item_name, COUNT(*) AS location_count
    FROM eldenringcatalog.gold.bridge_location_items
    GROUP BY item_name
    HAVING COUNT(*) = 1
)
SELECT 
    ic.item_name,
    l.location_name
FROM item_counts ic
INNER JOIN eldenringcatalog.gold.bridge_location_items bli ON ic.item_name = bli.item_name
INNER JOIN eldenringcatalog.gold.dim_locations l ON bli.location_id = l.location_id
LIMIT 50;

item_name location_name Ailment Talisman Abandoned Ailing Village Greater Potentates Cookbook (10) Abandoned Ailing Village Incursion Painting Artist's Shack (Gravesite Plain) Shadow Realm Rune 3 Belurat Gaol Greatjar Belurat Gaol Greater Potentates Cookbook (11) Belurat Gaol Demi-Human Swordsman Yosh Belurat Gaol Dried Bouquet Belurat Tower Settlement Stimulating Boluse Belurat Tower Settlement Bone Bow Belurat Tower Settlement Poisoned Hand Belurat Tower Settlement Divine Beast Head Belurat Tower Settlement Watchful Spirit Belurat Tower Settlement Ash of War: Shriek of Sorrow Belurat Tower Settlement Scorpion Liver Belurat Tower Settlement Horned Warriors Sword Belurat Tower Settlement Storeroom Key Belurat Tower Settlement Well Depths Key Belurat Tower Settlement Scorpion Stew Belurat Tower Settlement Tower of Shadow Message Belurat Tower Settlement Remembrance of the Dancing Lion Belurat Tower Settlement Rellanas Cameo Castle Ensis Shadow Realm Rune 2 Castle Ensis Shadow Realm Rune 4 Castle Ensis Drawstring Royal Magic Grease Castle Ensis Milady Castle Ensis Wolf Crest Shield Castle Ensis Moonrithylls Knight Sword Castle Ensis Carian Sorcery Sword Castle Ensis Glintstone Trio Castle Ensis Glintslab Fly Castle Ensis Dagger Castle Ensis Lazuli Robe Castle Ensis Heavy Crossbow Castle Ensis Royal Magic Grease Castle Ensis Message from Leda Castle Ensis Loyal Knights Cookbook Castle Ensis Remembrance of the Twin Moon Knight Castle Ensis Ash of War: Wing Stance Castle Ensis Monks Missive Castle Ensis May the Best Win Castle Ensis Cross Map Castle Ensis New Cross Map Castle Ensis Blessed Blue Dew Talisman Church of Benediction Oathseeker Knight Greaves Church of Benediction Oathseeker Knight Gauntlets Church of Benediction Oathseeker Knight Armor Church of Benediction Oathseeker Knight Helm Church of Benediction Pata Church of Benediction Dragon-Hunters Great Katana Dragon's Pit

# **📊 Question 26: Most common items across locations**

In [0]:
%sql
SELECT item_name, COUNT(*) AS location_count
FROM eldenringcatalog.gold.bridge_location_items
GROUP BY item_name
ORDER BY location_count DESC
LIMIT 30;

item_name location_count Stonesword Key 41 Rowa Fruit 39 Rune Arc 32 Mushroom 29 Furlcalling Finger Remedy 23 Golden Seed 22 Grave Violet 21 Smoldering Butterfly 20 Arteria Leaf 19 Trinas Lily 19 Erdleaf Flower 18 Herba 17 Nascent Butterfly 17 Smithing Stone (1) 16 Smithing Stone (7) 16 Golden Rune (1) 16 Golden Rune (10) 14 Silver Firefly 14 Sacred Tear 13 Smithing Stone (6) 13 Scadutree Fragment 13 Cave Moss 13 Budding Cave Moss 12 Smithing Stone (2) 12 Smithing Stone (8) 12 Root Resin 12 Smithing Stone (5) 12 Smithing Stone 4 12 Tarnished Golden Sunflower 12 Smithing Stone (4) 11

# **📊 28: Create exploration priority list**

In [0]:
%sql
WITH location_stats AS (
    SELECT 
        l.location_id,
        l.location_name,
        COALESCE(COUNT(DISTINCT bli.item_name), 0) AS item_count,
        COALESCE(COUNT(DISTINCT bln.npc_name), 0) AS npcs,
        COALESCE(COUNT(DISTINCT blb.boss_name), 0) AS bosses,
        COALESCE(COUNT(DISTINCT blc.creature_name), 0) AS creatures
    FROM eldenringcatalog.gold.dim_locations l
    LEFT JOIN eldenringcatalog.gold.bridge_location_items bli ON l.location_id = bli.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_npcs bln ON l.location_id = bln.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_bosses blb ON l.location_id = blb.location_id
    LEFT JOIN eldenringcatalog.gold.bridge_location_creatures blc ON l.location_id = blc.location_id
    GROUP BY l.location_id, l.location_name
)
SELECT 
    location_name,
    item_count,
    npcs,
    bosses,
    creatures,
    ROUND((item_count * 1.0) + (npcs * 10) + (bosses * 15) + (creatures * 0.3), 2) AS priority_score
FROM location_stats
ORDER BY priority_score DESC
LIMIT 20;

location_name item_count npcs bosses creatures priority_score Leyndell, Royal Capital 75 5 6 1 215.3 Volcano Manor 52 5 6 11 195.3 Stormveil Castle 82 4 4 8 184.4 Roundtable Hold 10 14 2 0 180.0 Raya Lucaria Academy 63 6 3 22 174.6 Seethewater Cave 18 13 1 6 164.8 Enir-Ilim 53 6 3 9 160.7 Shadow Keep 93 4 1 14 152.2 Bestial Sanctum 12 11 2 0 152.0 Leyndell, Ashen Capital 14 3 6 3 134.9 Mistwood 27 3 3 14 106.2 Castle Ensis 47 2 2 12 100.6 Siofra River 36 3 2 13 99.9 Belurat Tower Settlement 52 3 1 8 99.4 Ainsel River 29 1 4 1 99.3 Sealed Tunnel 13 7 1 4 99.2 Nokron, Eternal City 39 1 3 7 96.1 Redmane Castle 19 3 3 6 95.8 Cathedral of Manus Metyr 14 2 4 3 94.9 Stormhill 14 3 3 14 93.2